src/Util/EntityFieldQuery.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful\Util\EntityFieldQuery.
+ */
+
+namespace Drupal\restful\Util;
+
+use Drupal\restful\Exception\ServerConfigurationException;
+
+class EntityFieldQuery extends \EntityFieldQuery implements EntityFieldQueryRelationalConditionsInterface {
+
+  /**
+   * The relational filters.
+   *
+   * @var array[]
+   */
+  protected $relationships = array();
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRelationships() {
+    return $this->relationships;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addRelationship(array $relational_filter) {
+    $this->relationships[] = $relational_filter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function finishQuery($select_query, $id_key = 'entity_id') {
```

Comment:
this code is awesome and horrible at the same time :P. EFQ makes it so hard to alter it without prior knowledge of the actually query. You are a brave person :)